### ISSUE TITLE
Yarn causing to fail to build Docker container (ENOENT lstat)

### ISSUE THREAD

**Issue by icarito** [publiclab/plots2#3727]
Issue appears to be intermittent... 
```
Step 18/19 : RUN yarn --ignore-engines --ignore-scripts --modules-folder ./public/lib && yarn postinstall
 ---> Running in bc98ea39dc08
yarn install v1.10.1
[1/4] Resolving packages...
[91mwarning image-sequencer > ndarray-gaussian-filter > cwise > static-module > through2 > xtend > object-keys@0.4.0: 
[0m[2/4] Fetching packages...
[3/4] Linking dependencies...
[91merror An unexpected error occurred: "ENOENT: no such file or directory, lstat '/app/public/lib/acorn'".
[0minfo If you think this is a bug, please open a bug report with the information provided in "/app/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Service 'web' failed to build: The command '/bin/sh -c yarn --ignore-engines --ignore-scripts --modules-folder ./public/lib && yarn postinstall' returned a non-zero code: 1
Makefile:2: recipe for target 'build' failed
make: *** [build] Error 1
Build step 'Execute shell' marked build as failure
```

**Comment by jywarren** [comment]
Lot of similar looking errors in yarn repo https://github.com/yarnpkg/yarn/issues/4563

**Comment by icarito** [comment]
Yes I read them! And also tried a couple workarounds found there at https://github.com/publiclab/plots2/pull/3728/files

**Comment by jywarren** [comment]
Sorry, yeah, just wanting to document what we've tried, thanks!

**Comment by icarito** [comment]
Perhaps we can improve our Yarn deployment strategy taking some ideas from https://hackernoon.com/using-yarn-with-docker-c116ad289d56 - not sure if some version locking could help prevent issues like this.

**Comment by jywarren** [comment]
Hm, like including a `yarn.lock`?

**Comment by jywarren** [comment]
Oh, like the section on `Wait! Now Yarn is executed at each container build`?

Is this an intermittent issue or does the process always fail?

**Comment by jywarren** [comment]
Wondering if @Souravirus would be able to help figure this out, as he originally helped complete the Yarn transition work in #2659 ?

**Comment by Souravirus** [comment]
Yeah sure @jywarren, I would try my best to help

**Comment by Souravirus** [comment]
@jywarren yarn.lock is already included in our code

**Comment by icarito** [comment]
Here's another report from a similar issue with npm. I _guess_ this is a permissions issue, possibly coming from an npm package, and arising from mounting the `/app` directory as a volume. Perhaps it can be solved by adding a chmod or chown to Makefile.

**Comment by Radhikadua123** [comment]
While I was creating this fix: https://github.com/publiclab/plots2/pull/3775, I realised this issue. I'm able to reproduce it locally very well. I believe it's because of old yarn.lock file. May be we should commit that file with latest changes ? I can push the fix in that PR but I'm not sure how it will be tested in docker since I'm completely unaware of CI setup.

**Comment by Souravirus** [comment]
How did you reproduce this error locally @Radhikadua123?

**Comment by Radhikadua123** [comment]
@Souravirus 

```
rm -rf public
git checkout public
yarn parameters (it would give same error but there would be changes in the yarn.lock)
yarn parameters (Run it again and it won't give error. I think due to yarn.lock changes)
```

**Comment by jywarren** [comment]
Hi! Very cool, could you share a link to a branch on GitHub where youve
reproduced this and sourav can push it to our unstable container to see if
it gets solved there? Then we can try merging in your changes here.

**Comment by Souravirus** [comment]
Yeah sure. @Radhikadua123 please update your changes in one of your branches and share it here, so that I can try pushing it in unstable branch.

**Comment by Radhikadua123** [comment]
@jywarren @Souravirus Let me push them to https://github.com/publiclab/plots2/pull/3775

**Comment by Souravirus** [comment]
@jywarren updating the yarn.lock worked. Here is the link to the Jenkins result https://jenkins.laboratoriopublico.org/job/Plots-Unstable/322/console. Please see #3775.

**Comment by icarito** [comment]
Hi Radhika

Please yes, make the push for that commit and we can test it! Currently
we push to `unstable` branch and it automatically gets deployed to
https://unstable.publiclab.org/ - we would push your commits to that
branch and see whether it deploys correctly or not!

Thanks!

**Comment by icarito** [comment]
Oh, cool!

**Comment by Radhikadua123** [comment]
Fixed with #3775
